services/httpd/handler.go

```diff
@@ -292,14 +292,19 @@ func (h *Handler) serveQuery(w http.ResponseWriter, r *http.Request, user *meta.
 
 	// Make sure if the client disconnects we signal the query to abort
 	closing := make(chan struct{})
+	defer close(closing)
+
 	if notifier, ok := w.(http.CloseNotifier); ok {
 		notify := notifier.CloseNotify()
 		go func() {
-			<-notify
-			close(closing)
+			// Wait for either the request to close "closing"
+			// on exit, or for the http close notifier to kick in
+			select {
+			case <-closing:
+			case <-notify:
+				close(closing)
```

Comment:
10-4. I just realized that looking at the circleci tests. I've tried it a different way that should avoid the double close.